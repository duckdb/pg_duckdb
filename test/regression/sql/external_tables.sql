-- parquet
CREATE TABLE external_parquet () USING duckdb WITH (
    duckdb_external_location = '../../data/iris.parquet',
    duckdb_external_reader = 'read_parquet',
    duckdb_external_options = '{"filename": true, "file_row_number": true}'
);

SELECT "sepal.length", "file_row_number", "filename" FROM external_parquet ORDER BY "sepal.length", "file_row_number" LIMIT 5;
SELECT count(*) FROM external_parquet;

ALTER TABLE external_parquet RENAME TO external_parquet_renamed;
SELECT count(*) FROM external_parquet_renamed;

CREATE SCHEMA external_parquet_schema;
SET search_path to external_parquet_schema;
CREATE TABLE external_parquet () USING duckdb WITH (
    duckdb_external_location = '../../data/iris.parquet',
    duckdb_external_reader = 'read_parquet'
);
SELECT count(*) FROM external_parquet;
DROP TABLE external_parquet;
RESET search_path;

-- CSV
CREATE TABLE external_csv () USING duckdb WITH (
    duckdb_external_location = '../../data/iris.csv',
    duckdb_external_format = 'csv',
    duckdb_external_options = '{"header": true}'
);

SELECT count(*) FROM external_csv;
SELECT min("sepal.length"), max("sepal.length") FROM external_csv;

-- JSON
CREATE TABLE external_json () USING duckdb WITH (
    duckdb_external_location = '../../data/table.json',
    duckdb_external_format = 'json'
);

SELECT count(*) FROM external_json;
SELECT sum(a), min(b), max(c) FROM external_json;

-- CTAS from external table
CREATE TABLE json_tbl AS SELECT * FROM external_json;
SELECT count(*) FROM json_tbl;
SELECT sum(a), min(b), max(c) FROM json_tbl;

DROP TABLE external_parquet_renamed;
DROP TABLE external_csv;
DROP TABLE external_json;
DROP TABLE json_tbl;

SELECT count(*) FROM duckdb.external_tables;