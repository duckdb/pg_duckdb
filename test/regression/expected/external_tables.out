-- parquet
CREATE TABLE external_parquet () USING duckdb WITH (
    duckdb_external_location = '../../data/iris.parquet',
    duckdb_external_reader = 'read_parquet',
    duckdb_external_options = '{"filename": true, "file_row_number": true}'
);
SELECT "sepal.length", "file_row_number", "filename" FROM external_parquet ORDER BY "sepal.length", "file_row_number" LIMIT 5;
 sepal.length | file_row_number |        filename         
--------------+-----------------+-------------------------
          4.3 |              13 | ../../data/iris.parquet
          4.4 |               8 | ../../data/iris.parquet
          4.4 |              38 | ../../data/iris.parquet
          4.4 |              42 | ../../data/iris.parquet
          4.5 |              41 | ../../data/iris.parquet
(5 rows)

SELECT count(*) FROM external_parquet;
 count 
-------
   150
(1 row)

ALTER TABLE external_parquet RENAME TO external_parquet_renamed;
SELECT count(*) FROM external_parquet_renamed;
 count 
-------
   150
(1 row)

CREATE SCHEMA external_parquet_schema;
SET search_path to external_parquet_schema;
CREATE TABLE external_parquet () USING duckdb WITH (
    duckdb_external_location = '../../data/iris.parquet',
    duckdb_external_reader = 'read_parquet'
);
SELECT count(*) FROM external_parquet;
 count 
-------
   150
(1 row)

DROP TABLE external_parquet;
RESET search_path;
-- CSV
CREATE TABLE external_csv () USING duckdb WITH (
    duckdb_external_location = '../../data/iris.csv',
    duckdb_external_format = 'csv',
    duckdb_external_options = '{"header": true}'
);
SELECT count(*) FROM external_csv;
 count 
-------
   150
(1 row)

SELECT min("sepal.length"), max("sepal.length") FROM external_csv;
 min | max 
-----+-----
 4.3 | 7.9
(1 row)

-- JSON
CREATE TABLE external_json () USING duckdb WITH (
    duckdb_external_location = '../../data/table.json',
    duckdb_external_format = 'json'
);
SELECT count(*) FROM external_json;
 count 
-------
   100
(1 row)

SELECT sum(a), min(b), max(c) FROM external_json;
 sum  |  min   |  max  
------+--------+-------
 5050 | json_1 | 100.5
(1 row)

-- CTAS from external table
CREATE TABLE json_tbl AS SELECT * FROM external_json;
SELECT count(*) FROM json_tbl;
 count 
-------
   100
(1 row)

SELECT sum(a), min(b), max(c) FROM json_tbl;
 sum  |  min   |  max  
------+--------+-------
 5050 | json_1 | 100.5
(1 row)

DROP TABLE external_parquet_renamed;
DROP TABLE external_csv;
DROP TABLE external_json;
DROP TABLE json_tbl;
SELECT count(*) FROM duckdb.external_tables;
 count 
-------
     0
(1 row)

